### ISSUE TITLE
Php code generation for get methods - duplication

### ISSUE THREAD

**Issue by Nava2** [umple/umple#547]
Originally reported on Google Code with ID 649
Owned by `ellen.arteca`

---

Given class 

class X {}
class Y {
 3 ym_1 -- 1 X x;
}

the generated code in class Y contains the following (in php):

...

  public function getYm_1($index)
  {
    $aYm_1 = $this->ym_1[$index];
    return $aYm_1;
  }

  public function getYm_1()
  {
    $newYm_1 = $this->ym_1;
    return $newYm_1;
  }

...

this results in an error, since php function signatures are only dependent on their
names and not on the parameters - so, this results in an error in function declaration
(this is classified as a redeclaration).

---

Reported by `ellen.arteca` on 2014-10-28 19:32:30

**Comment by Nava2** [comment]
I think I've fixed this now... I changed the getMany jet file (for associations) to
generate `getYm_1_index($index)` and `getYm_1()` so the names are different.  However,
this resulted in compiler errors in other php tests where the original naming convention
was used, so I had to fix all those.

Should make a commit for this in the next few days (probably combined with a bigger
commit for issue 85).
I'll keep updating!

---

Reported by `ellen.arteca` on 2014-10-29 03:25:38

**Comment by Nava2** [comment]
So I made a commit for issue 85 which included a fix for this... this is revision 4678.
Like I explained above, I changed the getMany jet file for associations for php to
generate different names for the methods.  I also had to change a lot of the template
tests for php code generation because of this new syntax, along with various testbed
tests that reference it.

---

Reported by `ellen.arteca` on 2014-10-30 22:02:00
